pkg/gcpclient/client.go

```diff
@@ -0,0 +1,186 @@
+package gcpclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"golang.org/x/oauth2/google"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+	iam "google.golang.org/api/iam/v1"
+	"google.golang.org/api/option"
+)
+
+// Client is a wrapper object for actual GCP libraries to allow for easier mocking/testing.
+type Client interface {
+	// IAM
+	GetServiceAccount(accountName string) (*iam.ServiceAccount, error)
```

Comment:
Its a good practice not to use native object types in the interfaces. So anybody could implement them using standard types. But in this case this is gcp specific operator, we can ignore it :)